graveler/ref/manager.go

```diff
@@ -214,6 +214,7 @@ func (m *Manager) GetCommitByPrefix(ctx context.Context, repositoryID graveler.R
 					SELECT id, committer, message, creation_date, parents, meta_range_id, metadata
 					FROM graveler_commits
 					WHERE repository_id = $1 AND id >= $2
+					ORDER BY id
```

Comment:
I don't understand this change (but, to the PR's defence, I had several rounds with @ozkatz around this very SQL query).  This query asks for at most 2 rows; if there is more than one row then l.235 returns `ErrRefAmbiguous`.  So... AFAIU either `GetCommitByPrefix` fetches and returns a single row or it returns an error.  As such order *cannot* matter.

Reply:
The ordering is performed before the limit. Without the order by clause,
the query may return any 2 rows with keys greater than the prefix